src/Donations/resources/DonationDetails/app/components/DonorDetails/index.tsx

```diff
@@ -0,0 +1,141 @@
+import {useState} from 'react';
+import {__} from '@wordpress/i18n';
+import {useWatch} from 'react-hook-form';
+
+import SectionHeader, {DropdownTitle, HeaderLink} from '@givewp/components/AdminUI/SectionHeader';
+import {AsyncSelectDropdownField, DisabledTextField} from '@givewp/components/AdminUI/FormElements';
+import {FieldsetContainer} from '@givewp/components/AdminUI/ContainerLayout';
+import {useGetRequest} from '@givewp/components/AdminUI/api';
+import useDebounce from '@givewp/components/ListTable/hooks/useDebounce';
+
+import {StyleConfig} from './StyleConfig';
+import {apiNonce, apiRoot} from '../../../../window';
+
+/**
+ *
+ * @unreleased
+ */
+const endpoint = `${apiRoot.split('/donation')[0]}/donors`;
+const {donorId, firstName, lastName, email} = window.GiveDonations.donationDetails;
+
+export default function DonorDetails() {
+    const [dropdown, setDropdown] = useState<boolean>(true);
+
+    const handleDropdown = () => setDropdown(!dropdown);
+
+    return (
+        <section>
+            <SectionHeader>
+                <DropdownTitle isOpen={dropdown} title={__('Donor details', 'give')} handleDropdown={handleDropdown} />
+                <HeaderLink href={'/'}>{__('View donor details', 'give')}</HeaderLink>
+            </SectionHeader>
+            {dropdown && <SectionContainer />}
+        </section>
+    );
+}
+
+/**
+ *
+ * @unreleased
+ */
+export function SectionContainer() {
+    const [cachedDonors, setCachedDonors] = useState<CachedDonors>({
+        [donorId]: {
+            value: donorId,
+            label: `${firstName} ${lastName} (${email})`,
+            model: {
+                firstName,
+                lastName,
+                email,
+            },
+        },
+    });
+
+    const {getData} = useGetRequest(endpoint, apiNonce, '', '');
+    const watchedDonorId = useWatch({name: 'donorId'});
+    const currentDonor = cachedDonors[watchedDonorId].model;
+
+    const loadOptions = (inputValue: string, callback: (options: DonorOption[]) => void) => {
+        try {
+            const params = {
+                search: inputValue,
+                sortColumn: 'donorInformation',
+                sortDirection: 'asc',
+                perPage: 10,
+                return: 'model',
+            };
+            const queryString = Object.entries(params)
```

Comment:
Is it possible to use URLSearchParams to construct the queryString? I think it might help clean it up just a little.